### DISCUSSION TITLE
Is it Possible to use Email Instead of Username in OAuth2?

### DISCUSSION THREAD

**Discussion by rama-kairi** [fastapi/fastapi#3918] [Questions] [answered]
![image](https://user-images.githubusercontent.com/64131361/133885753-52902a62-1b84-4c41-b34d-2c24f6473b78.png)

Is it Possible to Use Email Instead of Username in OAuth2?

Here is a screenshot for reference.

**Comment by ghandic** [comment]
Absolutely, it would work in the same way the user name would though, unless you added additional checks the validate emails of your users.

Check out this tutorial where it is implemented for `user.name` you can just replace the logic for looking up by username for lookup by email and make email a mandatory field

https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/docs_src/security/tutorial003.py#L77

https://fastapi.tiangolo.com/tutorial/security/simple-oauth2/

If you also want to update the HTML to show "email" instead of "username" then I think you will need to create your own plugin for overriding the swagger layout (not part of the scope of FastAPI) 

https://swagger.io/docs/open-source-tools/swagger-ui/customization/custom-layout/

**Comment by YuriiMotov** [comment, marked as answer]
No, these names are defined by `OAuth2` standard.
If you change them it will mean you are not supporting `OAuth2` standard anymore, but your custom standard. So, 3rd party libraries will not work with your api.

But you can change labels in your UI
